### ISSUE TITLE
logs_assert contain logs from other tests

### ISSUE THREAD

**Issue by markcty** [dbrgn/tracing-test#18]
If the following tests run together, it will fail:
```rust
#[cfg(test)]
mod tests {
    use tracing::info;

    #[traced_test]
    #[test]
    fn no_log_from_other_test1() {
        info!("log count");
        logs_assert(|lines: &[&str]| {
            match lines
                .iter()
                .filter(|line| line.contains("log count"))
                .count()
            {
                1 => Ok(()),
                n => Err(format!("Count should be 1, but was {}", n)),
            }
        });
    }

    #[traced_test]
    #[test]
    fn no_log_from_other_test2() {
        info!("log count");
        logs_assert(|lines: &[&str]| {
            match lines
                .iter()
                .filter(|line| line.contains("log count"))
                .count()
            {
                1 => Ok(()),
                n => Err(format!("Count should be 1, but was {}", n)),
            }
        });
    }
}
```

I don't know if this is intentional. I created a PR to fix this problem.

**Comment by markcty** [comment]
See https://github.com/dbrgn/tracing-test/pull/19

**Comment by dbrgn** [comment]
Fixed in #19.
